base_layer/wallet_ffi/src/lib.rs

```diff
@@ -8955,7 +8955,7 @@ pub unsafe extern "C" fn wallet_is_recovery_in_progress(wallet: *mut TariWallet,
 #[no_mangle]
 pub unsafe extern "C" fn wallet_start_recovery(
     wallet: *mut TariWallet,
-    base_node_public_key: *mut TariPublicKey,
+    base_node_public_keys: *mut TariPublicKeys,
```

Comment:
We should not do this, 
We should flip this around. Currently it means the caller aka mobile needs to do the heavy lifting here. 

We should keep this a single base_node key. 
The logic inside the function should be:
```
if TariPublicKey == Null
{
     get all seed peers and use that
}
else{
     use only the specified peer
}
```